### PR TITLE
Upgrade to Laravel 7

### PR DESCRIPTION
**Pull request by rindibudiaramdhan**
# Change Dependency
- composer.json
- composer.lock
# Files Change to Adapt laravel 7.x
- app\Exceptions\Handler.php
- app\Http\kernel.php
- app\Http\Middleware\Cors.php
- config\app.php
- config\cors.php
- README.md
- resources\lang\en\passwords.php
- resources\lang\en\validation.php
- routes\channels.php
- routes\console.php
# Code Adapt to laravel 7.x
- $request->request->add() to $request->merge()
- $request->filled() to $request->has()
- use of the Response Enum
# Unit Test Coverage (Total 50.78%)
- AcceptanceReportTest
  ✓ get acceptance report
  ✓ store acceptance report
  ✓ store acceptance report fail item format
  ✓ get acceptance report by id
  ✓ get acceptance report statistic
  ✓ get acceptance report evidence
  ✓ get acceptance report detail
  ✓ get acceptance report realization logistic list
- ApplicantLetterTest  
  ✓ get applicant letter
  ✓ get search by letter number applicant letter
  ✓ get applicant letter by id
  ✓ store applicant letter
- AreaTest
  ✓ get cities
  ✓ get sub area
  ✓ get sub area with type area village
  ✓ get city total request
- IncomingLetterTest
  ✓ get incoming letter where mail status exists
  ✓ get incoming letter where mail status not exists
- LogisticRequestTest
  ✓ get logistic request no auth
  ✓ get export logistic request
  ✓ get logistic request list
  ✓ get logistic request by agency id no auth
  ✓ get unverified phase logistic request
  ✓ get recommendation phase logistic request
  ✓ get realization phase logistic request
  ✓ get finalized logistic request
  ✓ get rejected logistic request
  ✓ get logistic request filter
  ✓ get logistic request by agency id
  ✓ get logistic request by agency id not admin
  ✓ store logistic request faskes
  ✓ store logistic request non faskes
  ✓ logistic request needs
  ✓ logistic request summary
  ✓ post request verifying
  ✓ post request approval
  ✓ post request final
  ✓ post request set urgency
  ✓ post request return status from final phase
  ✓ post request return status from realization phase
  ✓ post request return status from recommendation phase
  ✓ post request return status from recommendation rejected phase
  ✓ post request return status from verification rejected phase
  ✓ put request update agency data
  ✓ put request update applicant data
  ✓ put request update letter data
  ✓ post upload logistic request letter
  ✓ post upload logistic request applicant file
- LogisticVerificationTest
  ✓ verification code registration
  ✓ verification resend code
  ✓ verification code registration not found
  ✓ verification confirmation fail
  ✓ verification confirmation success
- MasterFaskesTest
  ✓ get master faskes
  ✓ get master faskes by id
  ✓ store master faskes
  ✓ post verifying master faskes
  ✓ post rejecting master faskes
  ✓ get master faskes type list
  ✓ get faskes type total request
  ✓ get faskes type top request
- OutgoingLetterTest
  ✓ store outgoing letter
  ✓ get outgoing letter
  ✓ get outgoing letter by id
- ProductTest
  ✓ get products
  ✓ get product by id
  ✓ get product unit by id
  ✓ get products total request
  ✓ get products total request paginate
  ✓ get products top request
- TrackingTest
  ✓ get tracking
  ✓ get tracking by agency id
  ✓ get tracking by email
  ✓ get tracking by phone
  ✓ get tracking request phase items by agency id
  ✓ get tracking recommendation phase items by agency id
  ✓ get tracking finalization phase items by agency id
  ✓ get tracking outbound stogare list by agency id
  ✓ get trackingoutbound phase items by agency id
- UserTest
  ✓ post login success

image
image
image

## Issues Note:
- https://github.com/jabardigitalservice/pikobar-logistik-api/issues/233
- https://github.com/jabardigitalservice/pikobar-logistik-api/issues/250
- https://github.com/jabardigitalservice/pikobar-logistik-api/issues/232

### CODE REVIEW

**Review by ayocodingit — APPROVED** [review]

**Review by aldidoanta — APPROVED** [review]
Mantap kang @rindibudiaramdhan :+1: . Kalau bisa segera di-merge dan tes di staging, supaya bisa segera naik ke production. Supaya kalau ada penambahan fitur baru, bisa pakai standar Laravel 7

**Comment by rindibudiaramdhan** [comment]
siap kang @aldidoanta . Ini saya sekalian tambahkan Unit Test agar maintenancenya memudahkan overall pengujian.

**Comment by rindibudiaramdhan** [comment]
Disini saya menggunakan Github Action Laravel 7.x + MySQL 5.7 + PHP 7.3
Minus Pengecekan Codesniffer

**Comment by ayocodingit** [comment]
Kang @rindibudiaramdhan gercep banget upgrade laravel 5.8 ke laravel 7.0 😍, tapi mau bertanya dikit kang kenapa github action nya menggunakan MYSQL yah ? Ada kendala kah saat menggunakan SQLite

File: config/filesystems.php [hunk 1]
```diff
@@ -44,8 +44,12 @@
     'disks' => [
 
         'local' => [
-            'driver' => 'local',
-            'root' => storage_path('app'),
+            'driver' => 's3',
```
Comment (ayocodingit) [inline config/filesystems.php:47]: minor kang @rindibudiaramdhan bukannya untuk driver s3 udh ada tempatnya yah, sehingga tinggal ubah aja FILESYSTEM_DRIVER nya menjadi s3 :smile:

File: config/filesystems.php [hunk 2]
```diff
@@ -64,6 +68,12 @@
             'url' => env('AWS_CLOUDFRONT_URL'),
         ],
 
+        'testing' => [
```
Comment (ayocodingit) [inline config/filesystems.php:71]: btw kalau ini tujuannya buat apa yah kang @rindibudiaramdhan boleh di jelaskan

**Review by ayocodingit — COMMENTED** [review]
minor nya dikit kang @rindibudiaramdhan :smile:

**Comment by rindibudiaramdhan** [comment]
> Kang @rindibudiaramdhan gercep banget upgrade laravel 5.8 ke laravel 7.0 😍, tapi mau bertanya dikit kang kenapa github action nya menggunakan MYSQL yah ? Ada kendala kah saat menggunakan SQLite 😄

ada kendala kang @ayocodingit . ternyata format concat di SQLite berbeda dengan di MySQL.
Jika di MySQL menggunakan `CONCAT(<arg1>, <arg2>, ... , <argN>)` sedangkan di SQLite menggunakan `<arg1> || <arg2>`.
Sehingga mau tidak mau saya automated menggunakan MySQL

File: config/filesystems.php (hunk 1 above)
Comment (rindibudiaramdhan) [reply config/filesystems.php:47]: nah, baiknya seperti itu ya berarti?

File: config/filesystems.php (hunk 2 above)
Comment (rindibudiaramdhan) [reply config/filesystems.php:71]: ini untuk testing PHPUnit nya kang @ayocodingit .
bisa dilihat di file `phpunit.xml` menggunakan `<env name="FILESYSTEM_DRIVER" value="testing"/>`

File: config/filesystems.php (hunk 1 above)
Comment (ayocodingit) [reply config/filesystems.php:47]: iya kang bisa da filesystem_driver ubah jadi s3 nanti secara default akan ke s3

Comment (rindibudiaramdhan) [reply config/filesystems.php]: sip sudah ya kang @ayocodingit

File: config/filesystems.php (hunk 2 above)
Comment (ayocodingit) [reply config/filesystems.php:67]: oh iya ada fitur upload ke s3 yah, coba menggunakan mock kang @rindibudiaramdhan 

referensi: https://laracasts.com/discuss/channels/testing/unable-to-test-filesystem-temporaryurl
tapi aku belum coba, boleh di ulik kang :smile:

**Comment by ayocodingit** [comment]
> > Kang @rindibudiaramdhan gercep banget upgrade laravel 5.8 ke laravel 7.0 heart_eyes, tapi mau bertanya dikit kang kenapa github action nya menggunakan MYSQL yah ? Ada kendala kah saat menggunakan SQLite smile
> 
> ada kendala kang @ayocodingit . ternyata format concat di SQLite berbeda dengan di MySQL.
> Jika di MySQL menggunakan `CONCAT(<arg1>, <arg2>, ... , <argN>)` sedangkan di SQLite menggunakan `<arg1> || <arg2>`.
> Sehingga mau tidak mau saya automated menggunakan MySQL

oalah untuk query concat itu yah nice ilmu baru kang @rindibudiaramdhan :smile: 
selain karena query concat apakah ada lagi kang ?

File: config/filesystems.php (hunk 2 above)
Comment (rindibudiaramdhan) [reply config/filesystems.php:67]: yup sudah dicoba kok yang `Storage::fake();`
bisa dicek di `LogisticRequestTest`

**Comment by rindibudiaramdhan** [comment]
> > > Kang @rindibudiaramdhan gercep banget upgrade laravel 5.8 ke laravel 7.0 heart_eyes, tapi mau bertanya dikit kang kenapa github action nya menggunakan MYSQL yah ? Ada kendala kah saat menggunakan SQLite smile
> > 
> > 
> > ada kendala kang @ayocodingit . ternyata format concat di SQLite berbeda dengan di MySQL.
> > Jika di MySQL menggunakan `CONCAT(<arg1>, <arg2>, ... , <argN>)` sedangkan di SQLite menggunakan `<arg1> || <arg2>`.
> > Sehingga mau tidak mau saya automated menggunakan MySQL
> 
> oalah untuk query concat itu yah nice ilmu baru kang @rindibudiaramdhan 😄
> selain karena query concat apakah ada lagi kang ?

selain isu concat itu, belum dapat isu lain sih kang @ayocodingit 👍

**Pull request merged**